app.py

```diff
@@ -39,13 +40,27 @@ def initialize(cls, _db):
     _db = production_session()
     Conf.initialize(_db)
 
+def accepts_collection(f):
+    @wraps(f)
+    def decorated(*args, **kwargs):
+        collection = CollectionController(Conf.db).authenticated_collection_from_request()
+        if isinstance(collection, Response):
+            return collection
+        elif collection:
+            return f(collection=collection, *args, **kwargs)
```

Comment:
This isn't quite how `circulation` manages authentication -- instead of passing the patron into certain methods, it requests authentication details again inside the method itself. But because the code is arranged slightly differently and many of the controllers here are shared, this feels "good enough for now" without rearranging a bunch of the Metadata Wrangler's controllers.